packages/lockfile-lint-api/src/validators/ValidateIntegrity.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+module.exports = class ValidateIntegrity {
+  constructor ({packages} = {}) {
+    if (typeof packages !== 'object') {
+      throw new Error('expecting an object passed to validator constructor')
+    }
+
+    this.packages = packages
+  }
+
+  validate () {
+    let validationResult = {
+      type: 'success',
+      errors: []
+    }
+
+    for (const [packageName, packageMetadata] of Object.entries(this.packages)) {
+      if (!('integrity' in packageMetadata)) {
```

Comment:
should we show a warning if the flag was provided but some packages hold no integrity? I would think users would want to know

Reply:
integrity field is missing if you install from a GitHub URL. Not sure we need to verify those